astropy/utils/tests/test_diff.py

```diff
@@ -145,6 +145,17 @@ def test_tablediff():
     assert report_diff_values(a, a, fileobj=f)
 
 
+def test_large_table_diff():
+    # see https://github.com/astropy/astropy/issues/14010
+    colnames = [f"column{i}" for i in range(100)]
+    t1 = Table(names=colnames)
+
+    colnames.insert(50, "test")
+    t2 = Table(names=colnames)
+
+    assert not report_diff_values(t1, t2)
```

Comment:
Just to be tidy I'd suggest catching the output in a `StringIO()` object like above. It feels not nice to be printing some huge diff to stdout in the test.